### PR TITLE
chore: update gitlab-runner to be simpler and more in line with patterns

### LINKED ISSUE #46: Align gitlab-runner repos flavor specific values files to match that of other repos and composability with common zarf package and imports
### Describe what should be investigated or refactored
There is a pattern that is established for uds-packages where the flavor specific values (ie image refs) are explicitly defined in the <flavor>-values.yaml files in order to assist with renovate and make it easier to update flavors simultaneously. Additionally, we utilize zarf's composability to keep the component definitions DRY. We do this by creating a common component that gets imported into each flavored component. The common component also includes a common values file that holds the shared config values that are resued across the flavored components. See the uds-package-gitlab component and values definitions for a reference. 

Currently the gitlab-runner repo does not follow these patterns and the different flavored values have the entire config. Additionally there is not a common component that gets imported.

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This aligns the gitlab-runner deployment with other SWF patterns

## Related Issue

Fixes #46 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Other (security config, docs update, etc)

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-package-gitlab-runner/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: common/zarf.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/defenseunicorns/zarf/main/zarf.schema.json
+kind: ZarfPackageConfig
+metadata:
+  name: gitlab-runner
+  description: "UDS GitLab Runner Common Package"
+
+components:
+  - name: gitlab-runner
+    required: true
```
Comment (zachariahmiller) [inline common/zarf.yaml:9]: required: true does nothing here

Comment (zachariahmiller) [reply common/zarf.yaml:9]:
```suggestion
```

File: common/zarf.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    description: "Deploy gitlab-runner"
+    charts:
+      - name: uds-gitlab-runner-config
+        namespace: gitlab-runner
+        version: 0.1.0
+        localPath: ../chart
+        valuesFiles:
+          - ../values/common-values.yaml
```
Comment (zachariahmiller) [inline common/zarf.yaml:17]: config chart and main chart are using the same values?

File: zarf.yaml [hunk 3]
```diff
@@ -51,23 +31,15 @@ components:
...
-        namespace: gitlab-runner
-        url: https://charts.gitlab.io
-        gitPath: chart
-        version: "0.62.1"
         valuesFiles:
           - values/upstream-values.yaml
     images:
-      - "registry.gitlab.com/gitlab-org/gitlab-runner:alpine-v16.9.1"
-      - "library/alpine:latest"
+      - "registry.gitlab.com/gitlab-org/gitlab-runner:alpine-v16.9.1" # renovate: extractVersion=^alpine-(?<version>.*)$
```
Comment (zachariahmiller) [inline zarf.yaml:43]: 😢

Comment (zachariahmiller) [reply zarf.yaml:43]: # renovate: extractVersion=^alpine-(?<version>.*)$

File: zarf.yaml [hunk 4]
```diff
@@ -12,35 +12,15 @@ variables:
     default: gitlab-runner-sandbox
 
 components:
-  - name: gitlab-registration
```
Comment (zachariahmiller) [inline zarf.yaml:15]: 🥳

File: zarf.yaml (hunk 3 above)
Comment (Racer159) [reply zarf.yaml:43]: why tear?

File: common/zarf.yaml (hunk 2 above)
Comment (Racer159) [reply common/zarf.yaml:16]: yep!

Comment (Racer159) [reply common/zarf.yaml:16]: (we could not do that if you wanted)

File: zarf.yaml (hunk 3 above)
Comment (Racer159) [reply zarf.yaml:43]: https://f-droid.org/packages/com.phikal.regex/

Comment (zachariahmiller) [reply zarf.yaml:43]: having to use a comment

**Review by zachariahmiller — APPROVED** [review]
:shipit:  Looks good! 🥳

**Pull request merged**